### PR TITLE
[android] Add ability to override inferred MIME type when launching Intent

### PR DESCRIPTION
**Pull request by rhunt4675**
# Why

#1713 adds the ability to launch an intent with a Uri, however it would be nice to be able to explicitly specify a MIME type to accompany the Uri with [`setDataAndType`](https://developer.android.com/reference/android/content/Intent.html#setDataAndType(android.net.Uri,%20java.lang.String)) (Android tries to infer the MIME type based on the Uri but it is not always correct). This addresses an existing [feature request](https://expo.canny.io/feature-requests/p/need-ability-to-call-intentsettypemimetype-when-use-intentlauncherandroid).

# How

Added parameter to `startActivity` based on guidance from #1713. Caller can ignore new parameter to retain existing behavior. Planning to update the SDK once changes are merged.

### CODE REVIEW

**Comment by sjchmiela** [comment]
Hey, @rhunt4675! Thank you for your contribution! For the PR to be "ready to be merged", a couple of more places in the project will have to be adjusted to this change. 🙂 

- add fourth parameter `mime: string | null = null` to [`startActivityAsync` function](https://github.com/expo/expo/blob/a4598c6a439bcdfa03f7cd835929897a4e352f3d/packages/expo/src/IntentLauncherAndroid/IntentLauncherAndroid.ts#L108)
- pass that fourth parameter to [`ExponentIntentLauncher.startActivity`](https://github.com/expo/expo/blob/a4598c6a439bcdfa03f7cd835929897a4e352f3d/packages/expo/src/IntentLauncherAndroid/IntentLauncherAndroid.ts#L113)
- add information about `mime` argument to [the docs](https://github.com/expo/expo/blob/master/docs/versions/unversioned/sdk/intent-launcher.md)

Let me know if you have any questions!

**Comment by rhunt4675** [comment]
Oops, you're absolutely right! I was planning to make those changes to the expo-sdk repo, but didn't notice that it had been deprecated.

Thanks for the quick feedback. I believe I made the necessary changes, but let me know what else I can do to improve my PR.

File: docs/versions/unversioned/sdk/intent-launcher.md [hunk 1]
```diff
@@ -6,11 +6,19 @@ Provides a way to launch android intents. e.g. - opening a specific settings scr
...
-### `Expo.IntentLauncherAndroid.startActivityAsync(activity, data)`
+### `Expo.IntentLauncherAndroid.startActivityAsync(activity, data, uri, mime)`
 
-Starts the specified activity. The optional `data` parameter can be specified to pass additional data object to the activity. The method will return a promise which resolves when the user returns to the app.
+Starts the specified activity. The method will return a promise which resolves when the user returns to the app.
 
-There are a few pre-defined constants you can use for the `activity` parameter. You can find them at [expo/expo-sdk/src/IntentLauncherAndroid.js](https://github.com/expo/expo-sdk/blob/master/src/IntentLauncherAndroid.js).
+#### Arguments
+
+- **activity (_string_)** -- A string specifying which settings screen to open, or alternatively the action of the intent. There are a few pre-defined constants you can use for this parameter. You can find them at [expo/expo/src/IntentLauncherAndroid.js](https://github.com/expo/expo/blob/master/packages/expo/src/IntentLauncherAndroid/IntentLauncherAndroid.ts).
```
Comment (sjchmiela) [inline docs/versions/unversioned/sdk/intent-launcher.md]:
```suggestion
- **activity (_string_)** -- A string specifying which settings screen to open, or alternatively the action of the intent. There are a few pre-defined constants you can use for this parameter. You can find them at [expo/src/IntentLauncherAndroid.ts](https://github.com/expo/expo/blob/master/packages/expo/src/IntentLauncherAndroid/IntentLauncherAndroid.ts).
```

**Review by sjchmiela — COMMENTED** [review]
Thank you for this contribution! I left one small suggestion. 🙂 Could you also please add a [CHANGELOG](https://github.com/expo/expo/blob/master/CHANGELOG.md) entry under **master** – **New features**? Apart from that everything looks 💯!

**Comment by sjchmiela** [comment]
Also added suggestions to reformat types from `_string_` to:
```
`string`
```

It looks like Markdown compiler can't handle `_{ [key: string]: any }_` properly which is why `docs` CI failed (so my suggestion is to replace all `_type_` with backticks and call it a day. 🙂

**Comment by rhunt4675** [comment]
@sjchmiela Yep thanks! That fixed the `docs` CI checks. Let me know what else I can do to improve the PR.

**Review by sjchmiela — APPROVED** [review]
Looks great to me, thank you for your contribution!

**Pull request merged**